### DISCUSSION TITLE
gitlabci - won't update docker major version

### DISCUSSION THREAD

**Discussion by twarner-sofi** [renovatebot/renovate#9549] [Request Help]
**What Renovate type, platform and version are you using?**

Self hosted v24.114.2 pointed at gitlab.com

**Describe the bug**

When updating docker images in gitlab-ci files, it will not create a branch + MR for major verson updates. Minor version updates work as expected. It lists the update and branch in the logs but does not create the branch or push the change.

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
renovate-local    | DEBUG: packageFiles with updates (repository=twarner-sofi/renovate-test-3)
renovate-local    |        "config": {
renovate-local    |          "gitlabci": [
renovate-local    |            {
renovate-local    |              "packageFile": ".gitlab-ci.yml",
renovate-local    |              "deps": [
renovate-local    |                {
renovate-local    |                  "depName": "gradle",
renovate-local    |                  "currentValue": "6.7-jdk11",
renovate-local    |                  "replaceString": "gradle:6.7-jdk11",
renovate-local    |                  "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
renovate-local    |                  "datasource": "docker",
renovate-local    |                  "depType": "image",
renovate-local    |                  "depIndex": 0,
renovate-local    |                  "updates": [
renovate-local    |                    {
renovate-local    |                      "currentVersion": "6.7",
renovate-local    |                      "newVersion": "6.8",
renovate-local    |                      "newValue": "6.8-jdk11",
renovate-local    |                      "bucket": "non-major",
renovate-local    |                      "newMajor": 6,
renovate-local    |                      "newMinor": 8,
renovate-local    |                      "updateType": "minor",
renovate-local    |                      "isSingleVersion": true,
renovate-local    |                      "branchName": "renovate/docker-gradle-6.x"
renovate-local    |                    },
renovate-local    |                    {
renovate-local    |                      "currentVersion": "6.7",
renovate-local    |                      "newVersion": "7.0",
renovate-local    |                      "newValue": "7.0-jdk11",
renovate-local    |                      "bucket": "major",
renovate-local    |                      "newMajor": 7,
renovate-local    |                      "newMinor": 0,
renovate-local    |                      "updateType": "major",
renovate-local    |                      "isSingleVersion": true,
renovate-local    |                      "branchName": "renovate/docker-gradle-7.x"
renovate-local    |                    }
renovate-local    |                  ],
renovate-local    |                  "warnings": [],
renovate-local    |                  "fixedVersion": "6.7-jdk11"
renovate-local    |                }
renovate-local    |              ]
renovate-local    |            }
renovate-local    |          ]
renovate-local    |        }
renovate-local    | DEBUG: processRepo() (repository=twarner-sofi/renovate-test-3)
renovate-local    | DEBUG: Processing 1 branch: renovate/docker-gradle-6.x (repository=twarner-sofi/renovate-test-3)
```

</details>

**Have you created a minimal reproduction repository?**

https://gitlab.com/twarner-sofi/renovate-test-3

Expected behavior is 2 MRs, one for the 6.8 version and one for the 7.0

**Additional context**

**Comment by viceice** [comment]
Docker major updates are disabled by default, you need enable it manually until renovate [v25](#9394) has landed.

https://docs.renovatebot.com/docker/#enable-docker-major-updates

**Reply by attekemppila** [reply]
From user's point of view, this behavior isn't easily discoverable atm. Looking at discussions, this isn't the first time this has come up.

Sure, it's mentioned in [docker docs](https://docs.renovatebot.com/docker/) but for a non-standard behavior it's a bit [below the fold](https://docs.renovatebot.com/docker/#version-upgrading). tl;dr; And that assumes you realize to look for it there, once you are hit with this issue.

I mean, my config (extend [config:base](https://docs.renovatebot.com/presets-config/#configbase) + local stuff) handles other major updates just fine, so this is a bit unexpected.

Maybe, if there's major docker update detected, at least a simple INFO log would be nice: "Docker image xyz has major update. Docker major updates are disable by default at the moment. See: https://docs.renovatebot.com/docker/" Or something like that. Until behavior is changed in v25 (?).

Not a major issue, but still quality of life.

**Reply by rarkins** [reply]
V25 is now released

**Reply by attekemppila** [reply]
Aah... So, no longer an issue.

It seems that I'm still using 24.119.11 since that's the latest image: https://hub.docker.com/r/renovate/renovate/tags?page=1&ordering=last_updated&name=latest

Upgrading...
